version: '3.9'
services:
  readme.file-vault.postgres:
    image: postgres:14-alpine
    container_name: readme.file-vault.postgres
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - ${POSTGRES_PORT}:5432
    volumes:
      - readme.postgres.file-vault:/var/lib/postgresql/data

    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U ${POSTGRES_USER} -d {POSTGRES_PASSWORD}']
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s

    networks:
      - readme.file-vault

  readme.file-vault.pgadmin:
    image: dpage/pgadmin4
    container_name: readme.file-vault.pgadmin
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    ports:
      - ${PGADMIN_PORT}:80
    depends_on:
      - readme.file-vault.postgres
    networks:
      - readme.file-vault

volumes:
  readme.postgres.file-vault:

networks:
  readme.file-vault:
    driver: bridge
